gflownet/proxy/crystals/dave.py

```diff
@@ -76,19 +38,31 @@ def __init__(self, ckpt_path=None, release=None, rescale_outputs=True, **kwargs)
         self.scaled = False
 
         print("Initializing DAVE proxy:")
-        print("  Fetching proxy Github code...")
-        if not REPO_PATH.exists():
-            # creatre $root/external/repos
-            REPO_PATH.parent.mkdir(exist_ok=True, parents=True)
-            # clone remote proxy code
-            git.Repo.clone_from(REPO_URL, str(REPO_PATH))
-        # at this point the repo must exist
-        assert REPO_PATH.exists()
-        # checkout the appropriate tag/release
-        checkout_tag(release)
-
-        # import the proxu build funcion
-        sys.path.append(str(REPO_PATH))
+        print("  Checking out release:", release)
+
+        from importlib.metadata import PackageNotFoundError, version
```

Comment:
Why not have this import at the top of the file?

Reply:
This is a legacy lazy-loading pattern, I agree that now that the dave dependency has moved to the env and not the file system it can be moved up, thanks